targets/TARGET_Cypress/scripts/mbed_set_post_build_cypress.cmake

```diff
@@ -3,72 +3,90 @@
 
 include(mbed_set_post_build)
 
-set(MBED_POST_BUILD_CYPRESS_DIR "${CMAKE_CURRENT_LIST_DIR}")
-
 #
 # Merge Cortex-M4 HEX and a Cortex-M0 HEX.
 #
-function(mbed_post_build_psoc6_merge_hex mbed_target_name)
-    find_package(Python3)
+macro(mbed_post_build_psoc6_merge_hex cypress_psoc6_target)
+    if("${cypress_psoc6_target}" STREQUAL "${MBED_TARGET}")
+        function(mbed_post_build_function target)
+            find_package(Python3)
 
-    # Copy ${ARGN} to a variable first as it cannot be used directly with
-    # the list() command
-    set (extra_macro_args ${ARGN})
+            # Copy ${ARGN} to a variable first as it cannot be used directly with
+            # the list() command
+            set (extra_macro_args ${ARGN})
 
-    # Get the number of arguments past the last expected argument
-    list(LENGTH extra_macro_args num_extra_args)
+            # Get the number of arguments past the last expected argument
+            list(LENGTH extra_macro_args num_extra_args)
 
-    if(${num_extra_args} GREATER 0)
-        # Get extra argument as `cortex_m0_hex`
-        list(GET extra_macro_args 0 cortex_m0_hex)
-        set(post_build_command
-            COMMAND ${Python3_EXECUTABLE} ${MBED_POST_BUILD_CYPRESS_DIR}/PSOC6.py
-                merge
-                --elf ${CMAKE_BINARY_DIR}/$<TARGET_PROPERTY:mbed-post-build-bin-${mbed_target_name},application>.elf
-                --m4hex ${CMAKE_BINARY_DIR}/$<TARGET_PROPERTY:mbed-post-build-bin-${mbed_target_name},application>.hex
-                --m0hex ${cortex_m0_hex}
-        )
-    else()
-        set(post_build_command
-            COMMAND ${Python3_EXECUTABLE} ${MBED_POST_BUILD_CYPRESS_DIR}/PSOC6.py
-                merge
-                --elf ${CMAKE_BINARY_DIR}/$<TARGET_PROPERTY:mbed-post-build-bin-${mbed_target_name},application>.elf
-                --m4hex ${CMAKE_BINARY_DIR}/$<TARGET_PROPERTY:mbed-post-build-bin-${mbed_target_name},application>.hex
-        )
-    endif()
+            if(${num_extra_args} GREATER 0)
+                # Get extra argument as `cortex_m0_hex`
+                list(GET extra_macro_args 0 cortex_m0_hex)
```

Comment:
@rwalton-arm commented that the use of `${ARGN}` to get optional `--m0hex` image path is not rigorous. While it's a preexisting issue and not introduced by this PR, please let me know if we want to improve it here.

Reply:
Seems small enough to just fix rather than go through the overhead of filing and prioritizing an issue. Worth a separate commit in this PR to fix.

Reply:
I'll fix it

Reply:
I actually need some clarification, so made a reply to his original question: https://github.com/rwalton-arm/mbed-os/pull/1/files/eb17ce40d448eaee31fd6abd1963a11c439a05a5#r674938299

Reply:
Done, added a commit